tests/components/teslemetry/test_sensor.py

```diff
@@ -40,3 +48,56 @@ async def test_sensors(
     await hass.async_block_till_done()
 
     assert_entities_alt(hass, entry.entry_id, entity_registry, snapshot)
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_sensors_streaming(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    freezer: FrozenDateTimeFactory,
+    mock_vehicle_data: AsyncMock,
+    mock_listen: AsyncMock,
+) -> None:
+    """Tests that the sensor entities with streaming are correct."""
+
+    freezer.move_to("2024-01-01 00:00:00+00:00")
+
+    entry = await setup_platform(hass, [Platform.SENSOR])
+
+    # Stream update
+    runtime_data: TeslemetryData = entry.runtime_data
+    event = {
+        "vin": VEHICLE_DATA_ALT["response"]["vin"],
+        "data": {
+            Signal.DETAILED_CHARGE_STATE: "DetailedChargeStateCharging",
+            Signal.BATTERY_LEVEL: 90,
+            Signal.AC_CHARGING_ENERGY_IN: 10,
+            Signal.AC_CHARGING_POWER: 2,
+            Signal.CHARGING_CABLE_TYPE: None,
+            Signal.TIME_TO_FULL_CHARGE: 10,
+            Signal.MINUTES_TO_ARRIVAL: None,
+        },
+        "createdAt": "2024-10-04T10:45:17.537Z",
+    }
+    for listener, filters in runtime_data.vehicles[0].stream._listeners.values():
+        if recursive_match(filters, event):
+            listener(event)
```

Comment:
Let's not touch internals like entry.runtime_data. We should also not call listeners like this. Can we use the mock to get the registered functions and execute those?

Reply:
Done, ive changed the mock to contain the relevant code required to test this functionality.